Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2019-2020 Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleHistogram.H"
+#include "WarpX.H"
+#include "Utils/WarpXConst.H"
+#include "Utils/WarpXUtil.H"
+#include <AMReX_REAL.H>
+#include <AMReX_ParticleReduce.H>
+#include <limits>
+
+using namespace amrex;
+
+// constructor
+ParticleHistogram::ParticleHistogram (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+
+    // read parameters
+    ParmParse pp(rd_name);
+    std::string selected_species_name;
+    pp.get("species",selected_species_name);
+    pp.get("bin_number",m_bin_num);
+    pp.get("bin_max",   m_bin_max);
+    pp.get("bin_min",   m_bin_min);
+    pp.query("normalization",m_norm);
+    m_bin_size = (m_bin_max - m_bin_min) / m_bin_num;
+    std::string function_string = "";
+    Store_parserString(pp,"histogram_function(t,x,y,z,ux,uy,uz)",
+                       function_string);
+    m_parser.reset(new ParserWrapper<7>(
+        makeParser(function_string,{"t","x","y","z","ux","uy","uz"})));
+
+    // get MultiParticleContainer class object
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+    // get species names (std::vector<std::string>)
+    auto const species_names = mypc.GetSpeciesNames();
+    // select species
+    for ( int i = 0; i < mypc.nSpecies(); ++i )
+    {
+        if ( selected_species_name == species_names[i] )
+        { m_selected_species_id = i; }
```

Comment:
```suggestion
        if ( selected_species_name == species_names[i] ){
            m_selected_species_id = i;
        }
```
Could you please use standard indentation?